### PR TITLE
Fix diff_test not working for files in directories starting with "ext…

### PR DESCRIPTION
**Pull request by jesseschalken**
…ernal"

Example "external-dependencies/foo.txt"

### CODE REVIEW

**Comment by kormide** [comment]
Thanks for your submission. Are you able to sign the CLA and add a test for this fix?

File: lib/private/diff_test.bzl [hunk 1]
```diff
@@ -177,8 +177,8 @@ exit /b 1
 set -euo pipefail
 F1="{file1}"
 F2="{file2}"
-[[ "$F1" =~ ^external/* ]] && F1="${{F1#external/}}" || F1="$TEST_WORKSPACE/$F1"
-[[ "$F2" =~ ^external/* ]] && F2="${{F2#external/}}" || F2="$TEST_WORKSPACE/$F2"
+[[ "$F1" =~ ^external/+ ]] && F1="${{F1#external/}}" || F1="$TEST_WORKSPACE/$F1"
+[[ "$F2" =~ ^external/+ ]] && F2="${{F2#external/}}" || F2="$TEST_WORKSPACE/$F2"
```
[the remark this answers is not in the document]
Comment (jesseschalken) [reply lib/private/diff_test.bzl:181]: `/*` is zero or more slashes. `/+` is one or more. Just `/` would probably work as well.

**Comment by jesseschalken** [comment]
> Are you able to add a test for this fix?

Not any time soon.

**Comment by kormide** [comment]
I wrote this test but I can't reproduce your original issue. Is there something I'm missing?

```
load("//lib:diff_test.bzl", "diff_test")
load("@bazel_skylib//rules:copy_file.bzl", "copy_file")

copy_file(
    name = "foo_copy",
    src = "external-dir/foo.txt",
    out = "foo_copy.txt",
)

diff_test(
    name = "file_in_dir_with_external_prefix",
    file1 = "external-dir/foo.txt",
    file2 = "foo_copy.txt",
)
```

**Comment by jesseschalken** [comment]
@kormide Thanks!

**Pull request closed without merging**